src/Aspire.Dashboard/Model/ResourceOutgoingPeerResolver.cs

```diff
@@ -132,11 +132,40 @@ bool TryMatchResourceAddress(string value, [NotNullWhen(true)] out string? name,
         {
             foreach (var (resourceName, resource) in resources)
             {
+                // Try to match against URL endpoints
                 foreach (var service in resource.Urls)
                 {
                     var hostAndPort = service.Url.GetComponents(UriComponents.HostAndPort, UriFormat.UriEscaped);
 
-                    if (string.Equals(hostAndPort, value, StringComparison.OrdinalIgnoreCase))
+                    if (DoesAddressMatch(hostAndPort, value))
+                    {
+                        name = ResourceViewModel.GetResourceName(resource, resources);
+                        resourceMatch = resource;
+                        return true;
+                    }
+                }
+
+                // Try to match against connection strings using comprehensive parsing
+                if (resource.Properties.TryGetValue(KnownProperties.Resource.ConnectionString, out var connectionStringProperty) &&
```

Comment:
Previously the only data used for resolving peers was URLs. Now some properties are used as well. There is logic in this type for alerting listeners that resource data has changed and they need to recompute cached data using the resolver. At the moment it just looks for URLs changing. It needs to be extended to also check whether properties used here have changed.